acceptancetests/assess_primary_sub_relations.py

```diff
@@ -0,0 +1,225 @@
+    """
+
+    for _ in until_timeout(timeout):
+        current_status = client.get_status()
+        current_status.get_applications().keys()
+        if application_name not in current_status.get_applications().keys():
+            return
+        else:
+            time.sleep(1)
+    raise JujuAssertionError(
+        'Application "{}" failed to be removed in {} seconds'.format(
+            application_name, timeout))
+
+
+def assert_subordinate_is_removed(
+        client, primary_name, subordinate_name, timeout=60):
+    for _ in until_timeout(timeout):
+        status = client.get_status()
+        subordinate_names = [
+            unit[0] for unit in status.service_subordinate_units(
+                primary_name)
+            ]
+        if not any(subordinate_name in unit for unit in subordinate_names):
+            return
+        else:
+            time.sleep(1)
+    raise JujuAssertionError(
+        'Subordinate "{}" failed to be removed from "{}" in {} seconds'.format(
+            subordinate_name, primary_name, timeout))
+
+
+def create_model_with_primary_and_sub_apps(client, model_name):
+    test_model = client.add_model(model_name)
+    deploy_primary_charm(test_model, 'dummy-source')
+    deploy_primary_charm(test_model, 'dummy-sink')
+
+    test_model.deploy('cs:ntp')
+
+    test_model.set_config('dummy-source', {'token': 'subordinate-test'})
+    test_model.juju('add-relation', ('dummy-source', 'dummy-sink'))
+    test_model.juju('add-relation', ('ntp', 'dummy-source'))
+    test_model.juju('add-relation', ('ntp', 'dummy-sink'))
+    test_model.wait_for_workloads()
+    return test_model
+
+
+def ensure_subordinate_unit_operational(client, primary_app_name):
+    """Check status of subordinate charm on host.
+
+    Determine subordinate unit name and run the command 'service status' for it.
+
+    :raises JujuAssertionError: When service status exits non-zero.
+    """
+    sub_unit_name = get_subordinate_unit_name(client, primary_app_name)
+    primary_unit = '{}/0'.format(primary_app_name)
+    try:
+        client.juju(
+            'ssh',
+            (primary_unit, 'sudo', 'service', sub_unit_name, 'status'))
+    except subprocess.CalledProcessError:
+        raise JujuAssertionError('Subordinate charm unit not running.')
+
+
+def get_subordinate_unit_name(client, primary_app_name):
+    """get unit service name (jujud-unit-<name>-<number>)"""
+    all_subordinates = client.get_status().service_subordinate_units(
+        primary_app_name)
+    try:
+        # Get the first subordinate found.
+        sub_unit_name = [unit[0] for unit in all_subordinates][0]
+    except IndexError as e:
+        raise JujuAssertionError(
+            'Unable to find subordinates, key not found: {}'.format(e))
+    return 'jujud-unit-{}'.format(sub_unit_name.replace('/', '-'))
+
+
+def deploy_primary_charm(client, charm_name):
+    charm_path = local_charm_path(
+        charm=charm_name, juju_ver=client.version, series='xenial')
```

Comment:
series probably could/should be passed in (it's a default arg I think), with xenial as a reasonable default.